base/random.jl

```diff
@@ -1166,6 +1166,30 @@ immutable UUID
 end
 UUID(u::AbstractString) = convert(UUID, u)
 
+function uuid1(rng::AbstractRNG=GLOBAL_RNG)
+    u = rand(rng, Uint128)
+
+    # mask off clock sequence and node
+    u &= 0x00000000000000003fffffffffffffff
+
+    # set the unicast/multicast bit
+    u |= 0x00000000000000000000010000000000
+
+    # 0x01b21dd213814000 is the number of milliseconds between
+    # the UUID epoch and Unix epoch
+    timestamp = round(Uint64, time() * 1e7) + 0x01b21dd213814000
```

Comment:
However, it does look like all of the uses of `clock_now()` in `src/*.c` are to compute elapsed times within the process, so all of them should probably be using `uv_hrtime`.

That frees us to change `clock_now` to return a `uint64_t`.  On the Julia side, the conversion to double can happen in `time()`.